var/spack/repos/builtin/packages/qmcpack/package.py

```diff
@@ -155,6 +203,12 @@ def cmake_args(self):
         elif '~mpi' in spec:
             args.append('-DQMC_MPI=0')
 
+        # Default is parallel collective I/O enabled
+        if '+phdf5' in spec:
+            args.append('-DENABLE_PHDF5=1')
+        elif '~phdf5' in spec:
```

Comment:
I would use `else` here, it is a binary variant. Same for other if-statements in the `package.py`.

Reply:
I can address this. At some point, I did 'if-else', but then decided that I wanted symmetry. I can switch back to avoid the 'elif' and put a comment in so its easier on the eyes to follow all the conditionals.

Reply:
Actually, now I remember why I was explicit with these conditionals. The defaults are a combination of variants which are set to True or False. If I am not explicit about default variants, will the concretizer merge the default spec into the `spec` variable?

For example, will `spec` obtained from here:
`spack install qmcpack`
actually contain all the defaults variants
`spack install qmcpack+phdf5+qe+mpi+timers+<etc.>`

Reply:
Yes, if you don't specify, Spack uses the defaults. Boolean variants are either True or False, there is no "unspecified" option.